tools/src/main/java/org/apache/kafka/trogdor/workload/SustainedConnectionWorker.java

```diff
@@ -0,0 +1,563 @@
+            this.nextUpdate = System.currentTimeMillis() + this.refreshRate;
+            this.inUse = false;
+        }
+
+        @Override
+        protected void closeQuietly() {
+            Utils.closeQuietly(this.consumer, "KafkaConsumer");
+            this.consumer = null;
+            this.activePartition = null;
+        }
+    }
+
+    private static class SustainedConnectionThrottle extends Throttle {
+        SustainedConnectionThrottle(int maxPerPeriod) {
+            super(maxPerPeriod, THROTTLE_PERIOD_MS);
+        }
+    }
+
+    public class MaintainLoop implements Runnable {
+        @Override
+        public void run() {
+            Throttle throttle = new SustainedConnectionThrottle(1);
+            try {
+                while (!doneFuture.isDone()) {
+                    Optional<SustainedConnection> currentConnection = SustainedConnectionWorker.this.findConnectionToMaintain();
+                    if (currentConnection.isPresent()) {
+                        currentConnection.get().refresh();
+                    } else {
+                        throttle.increment();
+                    }
+                }
+            } catch (Exception e) {
+                SustainedConnectionWorker.this.totalAborts.incrementAndGet();
+                SustainedConnectionWorker.log.error("Abort while maintaining sustained connections", e.getMessage());
+            }
+        }
+    }
+
+    private Optional<SustainedConnection> findConnectionToMaintain() {
+        synchronized (SustainedConnectionWorker.this) {
+            for (SustainedConnection connection : this.connections) {
+                if (connection.needsRefresh()) {
+                    connection.claim();
+                    return Optional.of(connection);
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    private class StatusUpdater implements Runnable {
+        @Override
+        public void run() {
+            try {
+                JsonNode node = JsonUtil.JSON_SERDE.valueToTree(
+                    new StatusData(
+                            SustainedConnectionWorker.this.totalProducerConnections.get(),
+                            SustainedConnectionWorker.this.totalProducerFailedConnections.get(),
+                            SustainedConnectionWorker.this.totalConsumerConnections.get(),
+                            SustainedConnectionWorker.this.totalConsumerFailedConnections.get(),
+                            SustainedConnectionWorker.this.totalMetadataConnections.get(),
+                            SustainedConnectionWorker.this.totalMetadataFailedConnections.get(),
+                            SustainedConnectionWorker.this.totalAborts.get(),
+                            System.currentTimeMillis()));
+                status.update(node);
+            } catch (Exception e) {
+                SustainedConnectionWorker.log.error("Abort while running StatusUpdater", e.getMessage());
```

Comment:
This won't log the error. We want to use the public void error(String msg, Throwable t); overload, so we need to change e.getMessage() -> e

Reply:
Done.